src/main.go

```diff
@@ -236,15 +237,22 @@ func setupLogger() {
 		fmt.Printf("cannot access current user data\n")
 		return
 	}
+
 	homePath := user.HomeDir
-	logFilePath := homePath + "/" + currFileName
-	file, err := os.OpenFile(logFilePath, os.O_RDWR|os.O_CREATE|os.O_APPEND, 0666)
+	logDirectoryPath := homePath + "/" + logDirectory
+	err = os.MkdirAll(logDirectoryPath, 0755)
+	if err != nil {
+		fmt.Printf("error creating log directory : %s\n", logDirectoryPath)
+		return
+	}
+	logFilePath := logDirectoryPath + "/" + currFileName
+	file, err := os.OpenFile(logFilePath, os.O_RDWR|os.O_CREATE|os.O_APPEND, 0644)
```

Comment:
change `0644` to `0444` to make the generated log **Read-only**

Reply:
Done.